packages/shared/src/components/squads/SquadJoinButton.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQueryClient } from 'react-query';
+import { Squad } from '../../graphql/sources';
+import { Button } from '../buttons/Button';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useToastNotification } from '../../hooks/useToastNotification';
+import { useLeaveSquad, useJoinSquad } from '../../hooks';
+import { labels } from '../../lib';
+
+type SquadJoinProps = {
+  className?: string;
+  squad: Squad;
+  joinText?: string;
+  leaveText?: string;
+};
+
+export const SquadJoinButton = ({
+  className,
+  squad,
+  joinText = 'Join squad',
+  leaveText = 'Leave squad',
+}: SquadJoinProps): ReactElement => {
+  const queryClient = useQueryClient();
+  const { displayToast } = useToastNotification();
+  const { user, showLogin } = useAuthContext();
+  const isCurrentMember = !!squad.currentMember;
+
+  const { mutateAsync: joinSquad, isLoading: isJoiningSquad } = useMutation(
+    useJoinSquad({ squad }),
+    {
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+    },
+  );
+
+  const { mutateAsync: leaveSquad, isLoading: isLeavingSquad } = useMutation(
+    useLeaveSquad({ squad }),
+    {
+      onSuccess: () => {
+        displayToast('👋 You have left the squad.');
+
+        queryClient.invalidateQueries(['squad', squad.handle]);
+        queryClient.invalidateQueries(['squadMembersInitial', squad.handle]);
```

Comment:
Do we actually need to invalidate, what's the catch of not doing it?

Reply:
So that this is updated, it is optional though yes. Don't have the especial preference here if its worth it, WDYT?

<img width="187" alt="image" src="https://github.com/dailydotdev/apps/assets/9803078/1562c324-f0b9-4999-a17a-b673d5566e1b">

Reply:
Ah I see, well let's do it.
Doesn't cost too much, was just wondering what actually would change 😅